server/bundler/extensions-loader.js

```diff
@@ -24,27 +24,30 @@ function generateExtensionsModuleString( reducerRequires ) {
 function getReducersSync( extensionDirs = [] ) {
 	return extensionDirs
 		.filter( ( extensionDir ) => fs.existsSync( getReducerPath( extensionDir ) ) )
-		.map( ( extensionDir ) => generateReducerRequireString( extensionDir ) );
+		.map( generateReducerRequireString );
 }
 
 function getExtensionsModuleSync( extensionDirs ) {
 	return generateExtensionsModuleString( getReducersSync( extensionDirs ) );
 }
 
-function getReducersAsync( extensionDirs = [] ) {
-	return new Promise( ( resolveAll ) => {
-		Promise.all(
-			extensionDirs.map( ( extensionDir ) => {
-				return new Promise( ( resolveOne ) => {
-					fs.access( getReducerPath( extensionDir ), fs.constants.F_OK, ( err ) => {
-						resolveOne( ( ! err ) && generateReducerRequireString( extensionDir ) );
-					} );
-				} );
-			} )
-		).then( ( reducerRequires ) => resolveAll( compact( reducerRequires ) ) );
+function pathExists( testPath ) {
+	return new Promise( ( resolve ) => {
+		fs.access( testPath, fs.constants.F_OK, ( err ) => resolve( ! err ) );
 	} );
 }
 
+function getReducersAsync( extensionDirs = [] ) {
+	const dirsWithReducers = Promise.all(
+		extensionDirs.map( ( extensionDir ) =>
+			pathExists( getReducerPath( extensionDir ) )
+				.then( ( exists ) => exists && extensionDir ) )
```

Comment:
>  since `extensionDirs` is just a directory listing and shouldn't contain falsy values?

Correct. What I mean is `exists ? extensionDir : false`. Just as in the original, the idea is to get a list of the dirs that match. The final promise then gets rid of the falsy values with `compact`, thus the name `dirsWithReducers`. :) (Unless I missed something.)